### PR TITLE
Added reading of reference files

### PR DESCRIPTION
**Pull request by vgiuffrida**
A reference file is a text file with a reference to the directory containing a complete set of segments.

Added the trigger of trollstalker messages based on reference files: 
When a reference file is found, read all the the data included in the referenced directory and send a trollstalker message for each data segment.

### CODE REVIEW

File: bin/trollstalker.py [hunk 1]
```diff
@@ -141,17 +141,66 @@ def process(self, event):
...
+                       SourcePath = /path/to/dataset
+                       FileName = ref_file_filename
+                    """
+                    LOGGER.info("Found ref file: {}".format(file_to_check))
+                    try:
+                        reffile = RawConfigParser()
+                        reffile.read(file_to_check)
+                    except:
+                        LOGGER.error("Wrong ref file format: " + str(file_to_check))
+                        pass
```
Comment (mraspaud) [inline bin/trollstalker.py]: So even if the file format is wrong, the function continues ?!?

File: bin/trollstalker.py [hunk 2]
```diff
@@ -141,17 +141,66 @@ def process(self, event):
...
-                    LOGGER.info("Publishing message %s", str(message))
-                    self.pub.send(str(message))
+
+            # check if event is triggered on a ref file
+            #   if filesize<1kb it is a ref file
+            file_to_check = event.pathname
+            try:
+                file_check_size = os.path.getsize(file_to_check)
+                ref_file_parse = self.file_parser.parse(file_to_check)
+                if(file_check_size<=1000):
```
[the remark this answers is not in the document]
Comment (vgiuffrida) [reply bin/trollstalker.py]: Added "ref_filepattern" config option to describe the filename pattern of a reference file

File: bin/trollstalker.py [hunk 3]
```diff
@@ -141,17 +141,66 @@ def process(self, event):
...
+                    # if event is on a ref file, read inside the referenced directories
+                    """REF file internal format:
```
[the remark this answers is not in the document]
Comment (vgiuffrida) [reply bin/trollstalker.py]: moved to the docstring of the function

File: bin/trollstalker.py [hunk 4]
```diff
@@ -141,17 +141,66 @@ def process(self, event):
...
+                       [REF]
+                       SourcePath = /path/to/dataset
+                       FileName = ref_file_filename
+                    """
+                    LOGGER.info("Found ref file: {}".format(file_to_check))
+                    try:
+                        reffile = RawConfigParser()
+                        reffile.read(file_to_check)
+                    except:
```
[the remark this answers is not in the document]
Comment (vgiuffrida) [reply bin/trollstalker.py]: added exception type

File: bin/trollstalker.py (hunk 1 above)
Comment (vgiuffrida) [reply bin/trollstalker.py]: modified: now thread execution continues, but not the function execution.

**Pull request closed without merging**